galley/pkg/runtime/conversions/auth.go

```diff
@@ -0,0 +1,131 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package conversions
+
+import (
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	authn "istio.io/api/authentication/v1alpha1"
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/metadata"
+	"istio.io/istio/galley/pkg/runtime/resource"
+	"strings"
+)
+
+type ServiceToSelector struct {
+	Name map[string]resource.Labels
+}
+
+type AuthConverter struct {
+	Namespace map[string]ServiceToSelector
+}
+
+// AddService adds a service name and its associated selector labels to the AuthConverter.
+func (ac *AuthConverter) AddService(fullName resource.FullName, selector resource.Labels) {
+	if len(selector) == 0 {
+		// Ignore empty selector for stateless service.
+		// TODO: Figure out what is the best way to handle this.
+		return
+	}
+
+	namespace, name := fullName.InterpretAsNamespaceAndName()
+
+	if ac.Namespace == nil {
+		ac.Namespace = make(map[string]ServiceToSelector)
+	}
+
+	serviceToSelector, ok := ac.Namespace[namespace]
+	if !ok {
+		ac.Namespace[namespace] = ServiceToSelector{Name: map[string]resource.Labels{name: selector}}
+		return
+	}
+
+	if serviceToSelector.Name == nil {
+		serviceToSelector.Name = make(map[string]resource.Labels)
+	}
+	serviceToSelector.Name[name] = selector
+	return
+}
+
+// GetSelectors returns a list of selectors matched with the given service name and namespace.
+// If allowPrefixSuffix is true, allows "*" to be used in the name for prefix/suffix match, otherwise
```

Comment:
What is the usecase for allowPrefixSuffix

Reply:
For the service name used in authz policy.